src/app/tickets/page.tsx

```diff
@@ -1,41 +1,38 @@
 "use client"
 
 import { Button } from "@/components/ui/button"
-import { createTicket, getTicketsByUserId } from "@/services/ticket.service"
+import { api } from "@/trpc/react"
 import { useAuth } from "@clerk/nextjs"
-import { useEffect, useState } from "react"
+import { useCallback } from "react"
 
 const conferenceId = "67db25b2-0b76-4ab2-9c20-f73b16a5bb18"
 
 export default function Tickets() {
   const { isLoaded, userId } = useAuth()
-  const [tickets, setTickets] = useState<{ id: string }[]>([])
-  useEffect(() => {
-    if (!userId) return
-    getTicketsByUserId(userId)
-      .then((tickets) => {
-        setTickets(tickets)
-      })
-      .catch((error) => console.error(error))
+
+  const { data: tickets, refetch } = api.ticket.list.useQuery(
+    { userId: userId! },
+    { enabled: !!userId },
+  )
+
+  const apiCreateTicket = api.ticket.create.useMutation()
+  const createTicket = useCallback(async () => {
+    if (!userId) {
+      return
+    }
+    await apiCreateTicket.mutateAsync({ userId, conferenceId, state: "open" })
+    await refetch()
   }, [userId])
 
   if (!isLoaded || !userId) return null
 
   return (
     <main className="mx-auto w-full">
       <div className="text-center text-slate-400">
-        <Button
-          onClick={async () => {
-            await createTicket({
-              userId,
-              conferenceId,
-              state: "open",
-            })
-          }}>
-          Create ticket
-        </Button>
+        <Button onClick={() => createTicket()}>Create ticket</Button>
+        <Button onClick={() => refetch()}>Load tickets</Button>
       </div>
-      {tickets.map((ticket) => (
+      {tickets?.map((ticket) => (
```

Comment:
__Just thinking out loud__: I was wondering if it would make sense for us to make a atomic collection component 🤔
Basically, what it would do, is take in a list of literally anything (just has to be a list). It would then render its children if the list wasn't empty nor undefined. 

Otherwise it would print out a standard, muted, "no x found" <p> 🤔

Again, don't need to do that right now. Was just thinking out loud 😁